src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -0,0 +1,286 @@
+        Map<String, Object> sourceAndMetadataMap
+    ) {
+        IndexWrapper indexWrapper = new IndexWrapper(0);
+        Map<String, Object> result = new LinkedHashMap<>();
+        for (Map.Entry<String, Object> entry : knnMap.entrySet()) {
+            String targetKey = entry.getKey();
+            Object value = entry.getValue();
+            if (value instanceof String && StringUtils.isNotBlank(value.toString())) {
+                List<Float> modelTensor = modelTensorList.get(indexWrapper.index++);
+                result.put(targetKey, modelTensor);
+            } else if (value instanceof List) {
+                result.put(targetKey, processListOut((List<String>) value, modelTensorList, indexWrapper));
+            } else if (value instanceof Map) {
+                processMapTypeVectorOutput(targetKey, value, modelTensorList, indexWrapper, sourceAndMetadataMap);
+            }
+        }
+        return result;
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    private void processMapTypeVectorOutput(
+        String targetKey,
+        Object value,
+        List<List<Float>> modelTensorList,
+        IndexWrapper indexWrapper,
+        Map<String, Object> sourceAndMetadataMap
+    ) {
+        if (targetKey == null || sourceAndMetadataMap == null || value == null) return;
+        if (value instanceof Map) {
+            for (Map.Entry<String, Object> entry : ((Map<String, Object>) value).entrySet()) {
+                processMapTypeVectorOutput(
+                    entry.getKey(),
+                    entry.getValue(),
+                    modelTensorList,
+                    indexWrapper,
+                    (Map<String, Object>) sourceAndMetadataMap.get(String.valueOf(targetKey))
+                );
+            }
+        } else if (value instanceof String) {
+            sourceAndMetadataMap.put(targetKey, modelTensorList.get(indexWrapper.index++));
+        } else if (value instanceof List) {
+            sourceAndMetadataMap.put(targetKey, processListOut((List<String>) value, modelTensorList, indexWrapper));
+        }
+    }
+
+    private List<Map<String, List<Float>>> processListOut(
+        List<String> value,
+        List<List<Float>> modelTensorList,
+        IndexWrapper indexWrapper
+    ) {
+        List<Map<String, List<Float>>> numbers = new ArrayList<>();
+        for (String strValue : value) {
+            if (StringUtils.isNotBlank(strValue)) {
+                numbers.add(ImmutableMap.of(LIST_TYPE_NESTED_MAP_KEY, modelTensorList.get(indexWrapper.index++)));
+            }
+        }
+        return numbers;
+    }
+
+    private static void validateEmbeddingFieldsType(IngestDocument ingestDocument, Map<String, Object> embeddingFields) {
+        Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+        for (Map.Entry<String, Object> entry : embeddingFields.entrySet()) {
+            Object obj = sourceAndMetadataMap.get(entry.getKey());
```

Comment:
use better names for local variables, fix this throughout the PR. Having good variable names improves readability.